mutation_partition.cc

```diff
@@ -249,16 +249,16 @@ void mutation_partition::ensure_last_dummy(const schema& s) {
     }
 }
 
-void mutation_partition::apply(const schema& s, const mutation_partition& p, const schema& p_schema) {
-    apply_weak(s, p, p_schema);
+mutation_application_stats mutation_partition::apply(const schema& s, const mutation_partition& p, const schema& p_schema) {
+    return apply_weak(s, p, p_schema);
```

Comment:
Maybe this should be an output parameter. I don't think it's important enough to be the return value. @tgrabiec ?

It's good that you made mutation_application_stats a new type.

Reply:
(btw I usually dislike output parameters)

Reply:
I also think that stats are better collected by passing a stats collecting object as a parameter.

Reply:
Done, now `mutation_application_stats` is now passed as an output parameter. To make this parameter optional, I added overloads of `apply` and `apply_*` methods of `mutation_partition` that do not take this argument. Those variants use a temporary, stack-allocated `mutations_application_stats` structure.